lib/pascal.rb

```diff
@@ -1,36 +1,36 @@
+class PascalTriangle
+  attr_reader :first_row, :second_row, :space
+  def initialize
+    @first_row = [1]
+    @second_row = [1,1]
+    @space = " "
+  end
+  def draw(height)
+    num_spaces = height
+    for row in (1..height)
+      print_row(num_spaces, row)
+      num_spaces -= 1
+    end
+  end
 
-class Pascal
-
-
-  def nth_row(row_num=1)
-    row = [1,1]
-    new_row= []
+  private
+  def print_row(num_spaces, row)
+    puts "#{@space*num_spaces}#{nth_row(row).join(" ")}"
+  end
 
-    if row_num == 1
-      return "1"
-    elsif row_num == 2
-      return row.join(" ")
-    else
-      (row_num-2).times do
-        new_row = []
-        (row.length-1).times do |i|
-            new_row << (row[i]+row[i+1])
-        end
-        row = [1,1]
-        new_row.each do |num|
-          row.insert(1,num)
-        end
-      end
-      return row.join(" ")
-    end
+  def nth_row(row_num)
+    return @first_row  if row_num == 1
+    return @second_row if row_num == 2
+    row = @second_row
```

Comment:
Use `first_row` and `second_row` methods instead of these instance variables. Add an empty line after the last return statement.